tests/devices/qubit/test_simulate.py

```diff
@@ -1178,3 +1198,371 @@ def test_qinfo_tf(self):
+
+        if measure_f in (qml.sample,) and shots is None:
+            pytest.skip("Can't measure samples in analytic mode (`shots=None`)")
+
+        qs_with_mid_meas = qml.tape.QuantumScript(
+            [
+                qml.Hadamard(0),
+                qml.Hadamard(1),
+                qml.CZ([0, 1]),
+                qml.CNOT([0, 2]),
+                qml.CNOT([2, 3]),
+                qml.CZ([1, 3]),
+                qml.Toffoli([3, 2, 0]),
+                (m0 := qml.measure(0)).measurements[0],
+                qml.ops.op_math.Conditional(m0, qml.RZ(angles[0], 1)),
+                qml.Hadamard(1),
+                qml.Z(1),
+                (m1 := qml.measure(1)).measurements[0],
+                qml.ops.op_math.Conditional(m1, qml.RX(angles[1], 3)),
+            ],
+            [measure_f(wires=[0, 1, 2, 3])],
+            shots=shots,
+        )
+        qs_without_mid_meas = qml.tape.QuantumScript(
+            [
+                qml.Hadamard(0),
+                qml.Hadamard(1),
+                qml.CZ([0, 1]),
+                qml.CNOT([0, 2]),
+                qml.CNOT([2, 3]),
+                qml.CZ([1, 3]),
+                qml.Toffoli([3, 2, 0]),
+                qml.Hadamard(1),
+                qml.Z(1),
+                qml.RX(angles[1], 3),
+            ],
+            [measure_f(wires=[0, 1, 2, 3])],
+            shots=shots,
+        )  # approximate compiled circuit of the above
+        res1 = simulate_tree_mcm(qs_with_mid_meas, rng=rng)
+        res2 = simulate(qs_without_mid_meas, rng=rng)
+
+        if not isinstance(shots, list):
+            res1, res2 = (res1,), (res2,)
+
+        for rs1, rs2 in zip(res1, res2):
+            prob_dist1, prob_dist2 = rs1, rs2
+            if measure_f in (qml.sample,):
+                n_wires = rs1.shape[1]
+                prob_dist1, prob_dist2 = np.zeros(2**n_wires), np.zeros(2**n_wires)
+                for prob, rs in zip([prob_dist1, prob_dist2], [rs1, rs2]):
+                    index, count = np.unique(
+                        np.packbits(rs, axis=1, bitorder="little").squeeze(), return_counts=True
+                    )
+                    prob[index] = count
+
+            assert qml.math.allclose(
+                sp.stats.entropy(prob_dist1 + 1e-12, prob_dist2 + 1e-12), 0.0, atol=5e-2
+            )
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+    @pytest.mark.parametrize(
+        "postselect_mode", [None, "hw-like", "pad-invalid-samples", "fill-shots"]
+    )
+    def test_tree_traversal_interface_mcm(self, ml_framework, postselect_mode):
+        """Test that tree traversal works numerically with different interfaces"""
+        # pylint:disable = singleton-comparison, import-outside-toplevel
+        if ml_framework == "tensorflow":
+            import tensorflow as tf
+
+            tf.experimental.numpy.experimental_enable_numpy_behavior()
+
```

Comment:
Otherwise, the test for `tensorflow` interface fails because `ravel` can only be done when I enable `numpy` behavior. I didn't dig deep into what part of workflow was causing, but I suspect it using turning the `reset=True` for `m0`. I'll add a line for disabling at the end.

Reply:
We should probably not use `array.ravel` in that case. We can just use `qml.math.reshape` or `qml.math.ravel` wherever the error is coming from. I see this is a bug in the code rather than a bug in the test.

Reply:
Fixed it.